### ISSUE TITLE
[similar to #1033] broken 0.21.2 cannot load such file -- google/gax but needed by gcloud gem 

### ISSUE THREAD

**Issue by palladius** [GoogleCloudPlatform/google-cloud-ruby#1387]
I have a very similar issue as #1033 with gem google-cloud-error_reporting-0.21.2. 

Trying to install Google's GoogleCloudPlatform.getting-started-ruby/2-cloud-datastore/ and looking at this documentation: https://cloud.google.com/ruby/getting-started/using-cloud-datastore#installing_dependencies

$ bundle install

$ rails server
/var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1/error_group_service_api.rb:28:in require': cannot load such file -- google/gax (LoadError) from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1/error_group_service_api.rb:28:in<top (required)>'
from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1.rb:17:in require' from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1.rb:17:in<top (required)>'
from /var/lib/gems/2.3.0/gems/google-cloud-0.22.0/lib/google-cloud.rb:41:in require' from /var/lib/gems/2.3.0/gems/google-cloud-0.22.0/lib/google-cloud.rb:41:in<top (required)>'
from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:77:in require' from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:77:inblock (2 levels) in require'
from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:72:in each' from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:72:inblock in require'
from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:61:in each' from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:61:inrequire'
from /usr/lib/ruby/vendor_ruby/bundler.rb:99:in require' from /home/riccardo/git/gae-debiti/vendor/GoogleCloudPlatform.getting-started-ruby/2-cloud-datastore/config/application.rb:26:in<top (required)>'
from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:78:in require' from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:78:inblock in server'
from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:75:in tap' from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:75:inserver'
from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:39:in run_command!' from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands.rb:17:in<top (required)>'
from bin/rails:18:in require' from bin/rails:18:in

'
I've tried to uninstall that version of the gem as I have newer ones (0.23.3, 0.24.0), but apparently
google-cloud-0.22.0 depends on google-cloud-error_reporting (~> 0.21.0)

so I'm stuck on this library

**Comment by blowmage** [comment]
Hi @palladius, thanks for the issue. Unfortunately, this is completely unrelated to #1033. But, it is directly related to #1193 and #1194. The fix was provided in googleapis/gax-ruby#54.

Essentially, there are some incorrect file permissions set on a dependency that Error Reporting 0.21.2 has. And it is only an issue when the gem was installed by one user and is being used by another user. The user that owns the ruby process that is attempting to require that file does not have permissions to the file.

The first option is to uninstall all the gems and reinstall them using the same user that will be running them.

Another option is to upgrade to the latest gems. All gems were released on Friday and have compatible dependencies. Upgrade to the latest google-cloud and you will be all set.

The last option is to fix the gem's file permissions. To do this find the paths that the `google-gax` and `googleapis-common-protos` gems are installed in and to fix the file permissions.

```
$ bundle show google-gax
$ find <GOOGLE-GAX-PATH> -type f -perm 640 -print -exec chmod 644 {} \;
```

**Comment by palladius** [comment]
Oh wow- that was fast and effective. Looking fwd to try it out. Thanks!

**Comment by palladius** [comment]
So I ran this:

     sudo find $(bundle show google-gax) -type f -perm 640 -print -exec chmod 644 {} \;

and it fixed the error. Now I get a new one: 

$ rails server
/var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/devtools/clouderrorreporting/v1beta1/error_group_service_pb.rb:6:in `require': cannot load such file -- google/api/annotations_pb (LoadError)
	from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/devtools/clouderrorreporting/v1beta1/error_group_service_pb.rb:6:in `<top (required)>'
	from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1/error_group_service_api.rb:29:in `require'
	from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1/error_group_service_api.rb:29:in `<top (required)>'
	from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1.rb:17:in `require'
	from /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/lib/google/cloud/error_reporting/v1beta1.rb:17:in `<top (required)>'
	from /var/lib/gems/2.3.0/gems/google-cloud-0.22.0/lib/google-cloud.rb:41:in `require'
	from /var/lib/gems/2.3.0/gems/google-cloud-0.22.0/lib/google-cloud.rb:41:in `<top (required)>'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:77:in `require'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:77:in `block (2 levels) in require'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:72:in `each'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:72:in `block in require'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:61:in `each'
	from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:61:in `require'
	from /usr/lib/ruby/vendor_ruby/bundler.rb:99:in `require'
	from /home/riccardo/git/gae-debiti/vendor/GoogleCloudPlatform.getting-started-ruby/2-cloud-datastore/config/application.rb:26:in `<top (required)>'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:78:in `require'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:78:in `block in server'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:75:in `tap'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:75:in `server'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /var/lib/gems/2.3.0/gems/railties-4.2.6/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:18:in `require'
	from bin/rails:18:in `<main>'

Trying the same approach to directories:
- /var/lib/gems/2.3.0/gems/google-cloud-error_reporting-0.21.2/
- var/lib/gems/2.3.0/gems/google-cloud-0.22.0/

didn't fix it

**Comment by blowmage** [comment]
Right. Look at my comment again. You need to run that for both the `google-gax` and `googleapis-common-protos` gems. You should be good to go if you run that same command but for the protos gem.

**Comment by geigerj** [comment]
@palladius This should be fixed after the `0.8.4` release of `google-gax`, so closing. If you continue to have issues, please re-open or comment on this issue.
